crates/core/tedge_api/src/entity_store.rs

```diff
@@ -233,29 +285,35 @@ impl EntityStore {
     /// being registered.
     pub fn auto_register_entity(
```

Comment:
I've made an attempt to simplify it.

Reply:
Indeed simpler and clearer.

I think that we can go a step further by returning a vector of `RegistrationMessage` instead of `Mqtt::Message` (the latter can be easily built from the former by a mapper). => I will push a commit for that.

Reply:
I'll do that, as it'd allow the reuse of `try_convert_entity_registration` fn for both explicit registration as well as auto-registration.